core/src/test/scala/integration/kafka/api/AuthorizerIntegrationTest.scala

```diff
@@ -0,0 +1,360 @@
+//      val newTopic = "newTopic"
+//      val topicPartition = new TopicPartition(newTopic, 0)
+//      val newTopicResource = new Resource(Topic, newTopic)
+//      addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Read)), newTopicResource)
+//      try {
+//        this.consumers(0).assign(List(topicPartition).asJava)
+//        consumeRecords(this.consumers(0))
+//        Assert.fail("should have thrown exception")
+//      } catch {
+//        case e: TimeoutException =>
+//        //TODO need to update the consumer so if no metadata is propagated it throws server side exception instead of trying forever.
+//        case e: Exception => Assert.fail("Only timeout exception should be thrown.")
+//      }
+//
+//      addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Write)), newTopicResource)
+//      addAndVerifyAcls(Set(new Acl(KafkaPrincipal.ANONYMOUS, Allow, Acl.WildCardHost, Create)), Resource.ClusterResource)
+//      try {
+//        sendRecords(numRecords, topicPartition)
+//        consumeRecords(this.consumers(0))
+//      } catch {
+//        case e: Exception => Assert.fail("No exception should be thrown.")
+//      }
+//    }
+
+  def removeAllAcls() = {
+    authorizer.getAcls().keys.foreach { resource =>
+      authorizer.removeAcls(resource)
+      TestUtils.waitAndVerifyAcls(Set.empty[Acl], authorizer, resource)
+    }
+  }
+
+  def sendRequestAndVerifyResponseErrorCode(socket: Socket, key: Short, requestBytes: Array[Byte], expectedErrorCode: Short): Unit = {
+    sendRequest(socket, key, requestBytes)
+    val response = RequestKeyToResponseDeserializer(key)(receiveResponse(socket))
+    Assert.assertEquals(s"$key failed", expectedErrorCode, RequestKeyToErrorCode(key).asInstanceOf[(RequestOrResponse) => Short](response))
+  }
+
+  private def sendRequest(socket: Socket, id: Short, request: Array[Byte]) {
+    val outgoing = new DataOutputStream(socket.getOutputStream)
+    outgoing.writeInt(request.length + 2)
+    outgoing.writeShort(id)
+    outgoing.write(request)
+    outgoing.flush()
+  }
+
+  private def receiveResponse(socket: Socket): ByteBuffer = {
+    val incoming = new DataInputStream(socket.getInputStream)
+    val len = incoming.readInt()
+    val response = new Array[Byte](len)
+    incoming.readFully(response)
+    ByteBuffer.wrap(response)
+  }
+
+  private def sendRecords(numRecords: Int, tp: TopicPartition) {
+    val futures = (0 until numRecords).map { i =>
+      this.producers.head.send(new ProducerRecord(tp.topic(), tp.partition(), i.toString.getBytes, i.toString.getBytes))
+    }
+    try {
+      futures.foreach(_.get)
+    } catch {
+      case e: ExecutionException => throw e.getCause
+    }
+  }
+
+  private def addAndVerifyAcls(acls: Set[Acl], resource: Resource) = {
+    authorizer.addAcls(acls, resource)
+    TestUtils.waitAndVerifyAcls(authorizer.getAcls(resource) ++ acls, authorizer, resource)
```

Comment:
To really makes sure that the acl is propagated to the broker, we need to verify against the authorizer instance in the broker.